pkg/controller/job/job_controller_test.go

```diff
@@ -3736,8 +3735,7 @@ func TestSyncJobWithJobPodFailurePolicy(t *testing.T) {
 			},
```

Comment:
`wantStatusFailed` was not set... so was the test wrong before this change?

Reply:
Right, the intention of the code was to verify that for certain features Job controller waits with classifying terminating pods as Failed: https://github.com/kubernetes/kubernetes/blob/5e331569e4cd461baf19b9f2cc493af1f7f7a73f/pkg/controller/job/job_controller.go#L1878-L1885. Since PodDisruptionConditions is locked, we don't need two cases, and the only relevant feature gate is JobPodFailurePolicy, so I collapse the two test cases into: "terminating Pod not considered failed when JobPodFailurePolicy is enabled and used"